### PR TITLE
Use micrometer

### PR DESCRIPTION
**Pull request by filosganga**

### CODE REVIEW

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala [hunk 1]
```diff
@@ -1,61 +1,80 @@
...
+  ) {
+    def asDatadogConfig: DatadogConfig = {
+      val c = this;
+      new DatadogConfig {
+        override val apiKey = c.apiKey
+        override val applicationKey = c.applicationKey
+        override val enabled = true
+        override val step = java.time.Duration.ofSeconds(c.rate.toSeconds.toInt)
+        override val uri = c.host
+        def get(id: String): String = null
```
Comment (SystemFw) [inline src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: ??

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala [hunk 2]
```diff
@@ -1,61 +1,80 @@
...
-            new UdpTransport.Builder()
-              .withStatsdHost(c.host)
-              .build)
-          .withPrefix(c.prefix)
-          .build
-        r.start(c.rate._1, c.rate._2)
-        r
+  def create[F[_]](c: Config)(implicit F: Sync[F]): Resource[F, Reporter[F]] = {
+
+    val tf = new ThreadFactory {
```
Comment (SystemFw) [inline src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: is there no default for this?

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala [hunk 3]
```diff
@@ -1,61 +1,80 @@
...
+    }
 
-  def fromRegistry[F[_]](mx: MetricRegistry)(implicit F: Sync[F]): Reporter[F] =
+    for {
+      registry <- Resource.liftF(
+        F.delay(
+          DatadogMeterRegistry
+            .builder(c.asDatadogConfig)
+            .build()))
+      datadog = Resource.liftF(F.delay(registry.start(tf)))
```
Comment (SystemFw) [inline src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: is this done on purpose? as it stands I think it's a no op?

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala [hunk 4]
```diff
@@ -1,61 +1,80 @@
...
-          .withTags(c.tags.asJava)
-          .withTransport(
-            new UdpTransport.Builder()
-              .withStatsdHost(c.host)
-              .build)
-          .withPrefix(c.prefix)
-          .build
-        r.start(c.rate._1, c.rate._2)
-        r
+  def create[F[_]](c: Config)(implicit F: Sync[F]): Resource[F, Reporter[F]] = {
```
Comment (SystemFw) [inline src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: is this still a Resource? I don't see any closing action

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 1 above)
Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: The `DataDocConfig` is a java8 interface with some default implementations. It extends a set of interfaces and each of them calls `get(settingName)` that is abstract.

if get return null, they will use a default value that is usually hardcoded.

This is the reason of this mess.

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 2 above)
Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: Nope, there is a method with a default, but it is deprecated, the client needs to pass the `ThreadFactory`.

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 1 above)
Comment (SystemFw) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: Could you add a comment saying that returning `null` is required to trigger the correct defaulting behaviour in the underlying library?

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 2 above)
Comment (SystemFw) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: bah, ok

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 3 above)
Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: It is a mistake...but funny thing is that it is working :O

Comment (SystemFw) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: I would try to find out why... maybe you haven't suspended something you should have?

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 1 above)
Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: yep

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 3 above)
Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: I have figured out what happens: `start` is completely unnecessary as the constructor also starts it. however it needs to be stopped, so it is still a Resource. I have pushed this change.

File: src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala (hunk 4 above)
Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/monitor/metrics/Reporter.scala]: Yes as per the previous comment, the registry needs to be stopped.

**Pull request merged**